src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1674,6 +1674,8 @@ private BoundBlock BindBlockParts(BlockSyntax node, DiagnosticBag diagnostics)
                 boundStatements.Add(boundStatement);
             }
 
+            var immutableBoundStatements = boundStatements.ToImmutableAndFree();
```

Comment:
>immutableBoundStatements [](start = 16, length = 24)

Minor point: Consider renaming `boundStatements` to `builder` and naming this local `boundStatements` (or simply `statements`).